db/migrate/20170929220708_add_original_spot_to_user.rb

```diff
@@ -0,0 +1,57 @@
+class AddOriginalSpotToUser < ActiveRecord::Migration[5.0]
+  def up
+    add_column :users, :original_spot_id, :integer, index: true
+    add_foreign_key :users, :spots, column: :original_spot_id
+
+    User.performers.includes(:current_spot, :user_challenges, :challenges, :discipline_actions).each do |user|
+      user.original_spot = original_spot_for_user(user)
+      user.save
+    end
+  end
+
+  def down
+    remove_column :users, :original_spot_id
+  end
+
+  private
+
+  def original_spot_for_user(user)
+    has_discipline_action = user.discipline_actions.where(open_spot: true).count.positive?
+    been_in_a_challenge = user.user_challenges.count.positive?
+
+    # If a user's spot has never been opened and they've never been in a challenge,
+    # then their spot's never been changed
+    # If a user's spot has never been opened, but they've been in a challenge, then
+    # their first UserChallenge.spot is their original spot
+    # If user's spot has been opened, but they've never been in a challenge, there are two options
+    # - If the user isn't an alternate, then they haven't lost their spot yet, so use the current spot
+    # - If the user is an alternate, then they lost their spot. Their current spot was used as a UserChallenge.spot
+    # - in a challenge for the performance for which their spot was opened
+    # If the user's spot has been opened and they've been in a challenge, we'll have to do some more digging
+    # - If the user's first user_challenge was created before their first discipline action, then user the UserChallenge.spot
+    # - Else
+    if !has_discipline_action && !been_in_a_challenge
+      user.current_spot
+    elsif !has_discipline_action && been_in_a_challenge
+      user.user_challenges.order(id: :asc).first.spot
+    elsif has_discipline_action && !been_in_a_challenge && user.alternate?
+      performance = user.discipline_actions.join(:performances).where(open_spot: true).order(id: :asc).first.performance
+      challenge = Challenge.joins(:user_challenges).where('performance_id = ? AND user_challenges.spot_id = ?', performance.id, user.current_spot.id)
```

Comment:
Why the SQL here? 

`where(performance_id: performance.id, user_challenges: { spot_id: user.current_spot.id })` should handle that